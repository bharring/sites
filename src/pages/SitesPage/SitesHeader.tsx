import { IonButton, IonButtons, IonHeader, IonIcon, IonMenuButton, IonTitle, IonToolbar } from '@ionic/react';

export const SitesHeader: React.FC<{ title: string; icon: string; toggleView: () => void }> = ({ title, icon, toggleView }) => {
  return (
    <IonHeader>
      <IonToolbar>
        <IonButtons slot="start">
          <IonMenuButton></IonMenuButton>
        </IonButtons>
        <IonTitle>{title}</IonTitle>
        <IonButtons slot="primary">
          <IonButton data-testid="button" slot="end" onClick={toggleView}>
            <IonIcon icon={icon} />
          </IonButton>
        </IonButtons>
      </IonToolbar>
    </IonHeader>
  );
};

export default SitesHeader;
