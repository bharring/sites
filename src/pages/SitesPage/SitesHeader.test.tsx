import '@testing-library/jest-dom';

import { cleanup, fireEvent, render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { SitesHeader } from './SitesHeader';

describe('<SitesHeader />', () => {
  afterEach(cleanup);

  test('matches snapshot', () => {
    const toggleView = jest.fn();
    const icon = 'icon';
    const title = 'Page Title';
    const tree = renderer.create(<SitesHeader title={title} icon={icon} toggleView={toggleView} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders page title', () => {
    const toggleView = jest.fn();
    const icon = 'icon';
    const title = 'Page Title';
    const { queryByText } = render(<SitesHeader title={title} icon={icon} toggleView={toggleView} />);

    expect(queryByText('Page Title')).toBeTruthy();
  });

  test('calls toggle', () => {
    const toggleView = jest.fn();
    const icon = 'icon';
    const title = 'Page Title';
    const { getByTestId } = render(<SitesHeader title={title} icon={icon} toggleView={toggleView} />);

    expect(toggleView).not.toHaveBeenCalled();
    fireEvent.click(getByTestId('button'));
    expect(toggleView).toHaveBeenCalled();
  });
});
