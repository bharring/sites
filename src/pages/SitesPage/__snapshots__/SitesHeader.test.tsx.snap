// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SitesHeader /> matches snapshot 1`] = `
<ion-header>
  <ion-toolbar>
    <ion-buttons
      slot="start"
    >
      <ion-menu-button />
    </ion-buttons>
    <ion-title>
      Page Title
    </ion-title>
    <ion-buttons
      slot="primary"
    >
      <ion-button
        data-testid="button"
        onClick={[Function]}
        slot="end"
      >
        <ion-icon
          icon="icon"
        />
      </ion-button>
    </ion-buttons>
  </ion-toolbar>
</ion-header>
`;
